src/app/query/nonce/account-nonces.hooks.ts

```diff
@@ -0,0 +1,34 @@
+import { useGetAccountNonces } from '@app/query/nonce/account-nonces.query';
+import { useCurrentAccountFilteredMempoolTransactionsState } from '@app/query/mempool/mempool.hooks';
+
+export function useNextNonce() {
+  const { data: accountNonces } = useGetAccountNonces();
+  const pendingTransactions = useCurrentAccountFilteredMempoolTransactionsState();
+
+  if (!accountNonces) return 0;
+
+  let nextNonce = 0;
+
+  const missingNonces = accountNonces.detected_missing_nonces;
+  const possibleNextNonce = accountNonces.possible_next_nonce;
+
+  const hasMissingNonce = missingNonces?.length > 0;
+  const missingNonce = missingNonces?.sort()[0];
+
+  const hasPendingTx = pendingTransactions?.length > 0;
+  const latestPendingTx = pendingTransactions[0];
+  const pendingTxNonce = latestPendingTx?.nonce;
+
+  if (hasMissingNonce) {
+    if (hasPendingTx && missingNonce !== pendingTxNonce) {
+      nextNonce = Math.min(missingNonce, pendingTxNonce);
+    }
+    nextNonce = missingNonce;
+  } else if (hasPendingTx) {
+    nextNonce = Math.max(pendingTxNonce + 1, possibleNextNonce);
+  } else {
+    nextNonce = possibleNextNonce;
+  }
```

Comment:
This would be flatter, and easier to read with early returns imo:

along the format of, 

```ts
if (!isNumber(hasMixingNonce)) return missingNonce;

if (hasPendingTx && missingNonce !== pendingTxNonce) return Math.min(missingNonce, pendingTxNonce);

if (hasPendingTx) return Math.max(pendingTxNonce + 1, possibleNextNonce);

return possibleNextNonce;
```